### PR TITLE
Add nvidia support for `chicoma-cpu`

### LINKED ISSUE #182: Add chicoma-cpu nvidia support under spack

### PR DESCRIPTION
**Pull request by altheaden**
This PR adds the necessary `.yaml`, `.sh` and `.csh` files to add nvidia support for `chicoma-cpu` to mache. 

While creating these files, I noticed two small typos in `pm-cpu` files, so I included the fixes for those as well.

closes #182 

Checklist
* [ ] `Testing` comment in the PR documents testing used to verify the changes

### CODE REVIEW

**Comment by altheaden** [comment]
## Testing

I have not done any testing for this branch because I don't have access to chicoma.

**Comment by altheaden** [comment]
It is likely that some of the version numbers will need to change. I will take a look in the machine config file to see if there's anything I can update myself (this just occurred to me after creating the PR so that's why it's not currently included!)

**Comment by altheaden** [comment]
Just updated the version numbers slightly to what I inferred might be closer to what we need. I think it's very possible that I missed something somewhere, but hopefully this is enough to get things started for you @xylar

**Comment by xylar** [comment]
Great!  I'll take a look tomorrow and update the specific module hashes and version numbers but this is an excellent place for me to start from!

File: mache/spack/chicoma-cpu_nvidia_mpich.yaml [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      externals:
+      - spec: python@3.10.9
+        prefix: /usr/projects/hpcsoft/common/x86_64/anaconda/2023.03-python-3.10
+        modules:
+        - python/3.10-anaconda-2023.03
+      buildable: false
+    cray-mpich:
+      externals:
+      - spec: cray-mpich@8.1.26
+        prefix: /opt/cray/pe/mpich/8.1.25/ofi/nvidia/20.7
```
Comment (xylar) [inline mache/spack/chicoma-cpu_nvidia_mpich.yaml:94]:
```suggestion
        prefix: /opt/cray/pe/mpich/8.1.26/ofi/nvidia/20.7
```

File: mache/spack/chicoma-cpu_nvidia_mpich.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+export MPICH_VERSION_DISPLAY=1
+## purposefully omitting OMP variables that cause trouble in ESMF
+# export OMP_STACKSIZE=128M
+# export OMP_PROC_BIND=spread
+# export OMP_PLACES=threads
+export HDF5_USE_FILE_LOCKING=FALSE
+export PERL5LIB=/usr/projects/climate/SHARED_CLIMATE/software/chicoma-cpu/perl5-only-switch/lib/perl5
+export PNETCDF_HINTS="romio_ds_write=disable;romio_ds_read=disable;romio_cb_write=enable;romio_cb_read=enable"
+export MPICH_COLL_SYNC=MPI_Bcast
+export MPICH_GPU_SUPPORT_ENABLED=1
```
Comment (xylar) [inline mache/spack/chicoma-cpu_nvidia_mpich.sh:46]:
```suggestion
```
I don't think we want this one here.  It's probably missing from `chicoma-gpu` but that's a separate issue.

File: mache/spack/chicoma-cpu_nvidia_mpich.csh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+setenv MPICH_VERSION_DISPLAY 1
+## purposefully omitting OMP variables that cause trouble in ESMF
+# setenv OMP_STACKSIZE 128M
+# setenv OMP_PROC_BIND spread
+# setenv OMP_PLACES threads
+setenv HDF5_USE_FILE_LOCKING FALSE
+setenv PERL5LIB /usr/projects/climate/SHARED_CLIMATE/software/chicoma-cpu/perl5-only-switch/lib/perl5
+setenv PNETCDF_HINTS "romio_ds_write=disable;romio_ds_read=disable;romio_cb_write=enable;romio_cb_read=enable"
+setenv MPICH_COLL_SYNC MPI_Bcast
+setenv MPICH_GPU_SUPPORT_ENABLED 1
```
Comment (xylar) [inline mache/spack/chicoma-cpu_nvidia_mpich.csh:46]:
```suggestion
```
Again, probably don't want this.

**Review by xylar — COMMENTED** [review]
A few small changes.  I may go ahead and commit them so I can test.

File: mache/spack/chicoma-cpu_nvidia_mpich.yaml [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+      providers:
+        mpi: [cray-mpich@8.1.26]
+{% if e3sm_lapack %}
+        lapack: [cray-libsci@23.05.1.4]
+{% endif %}
+    bzip2:
+      externals:
+      - spec: bzip2@1.0.6
+        prefix: /usr
+      buildable: false
```
Comment (xylar) [inline mache/spack/chicoma-cpu_nvidia_mpich.yaml:28]:
```suggestion
    bzip2:
      externals:
      - spec: bzip2@1.0.6
        prefix: /usr
      buildable: false
    cmake:
      externals:
      - spec: cmake@3.25.1
        prefix: /usr/projects/hpcsoft/cos2/common/x86_64/cmake/3.25.1
      buildable: false
```

Comment (xylar) [reply mache/spack/chicoma-cpu_nvidia_mpich.yaml:28]: We will likely want to add this to `chicoma-cpu_gnu_mpich.yaml` as well.

**Review by xylar — COMMENTED** [review]
One more thing it seems like we need.  I'm not able to successfully build cmake (it's trying to build python-related thing for some reason).

**Review by xylar — APPROVED** [review]
## Testing

I was able to build the Polaris spack environments and build and successfully run Omega with them on Chicoma-CPU with nvidia.  I needed to use the Polaris branch from https://github.com/E3SM-Project/polaris/pull/211.

**Pull request merged**